toke.c

```diff
@@ -4428,30 +4428,30 @@ S_intuit_more(pTHX_ char *s, char *e)
     if (PL_lex_brackets)
         return TRUE;
 
-    if (    *s == '-'
+    if (    s[0] == '-'
         &&  s[1] == '>'
         && (s[2] == '[' || s[2] == '{'))
     {
         return TRUE;
     }
 
-    if (   *s == '-'
+    if (   s[0] == '-'
         && s[1] == '>'
         && FEATURE_POSTDEREF_QQ_IS_ENABLED
         && (   (s[2] == '$' && (    s[3] == '*'
                                 || (s[3] == '#' && s[4] == '*')))
             || (s[2] == '@' && memCHRs("*[{", s[3])) ))
         return TRUE;
 
-    if (*s != '{' && *s != '[')
+    if (s[0] != '{' && s[0] != '[')
         return FALSE;
 
     PL_parser->sub_no_recover = TRUE;
     if (!PL_lex_inpat)
         return TRUE;
 
     /* In a pattern, so maybe we have {n,m}. */
-    if (*s == '{') {
+    if (s[0] == '{') {
```

Comment:
I don't understand this.  regcurly had very restricted syntax, so the authors could have imagined that anything that matched that would necessarily mean a character class.  But the syntax for regcurly has been relaxed somewhat, and the absence of it being regcurly doesn't mean it can't be a character class

Reply:
Hmm, github attached my comment to the wrong line.

Reply:
addressed in commentary commit